server/utils/dateUtils.ts

```diff
@@ -153,6 +159,10 @@ export const dateAndTimeInputsAreValidDates = <K extends string | number>(
   dateInputObj: Partial<ObjectWithDateParts<K>>,
   key: K,
 ): boolean => {
+  const inputYear = dateInputObj?.[`${key}-year`] as string
+
+  if (inputYear && inputYear.length !== 4) return false
```

Comment:
Before this were 2 digit years valid (EG '24)? I'm not sure off the top of my head whether that would parse but if so we should probably retain that behaviour

Reply:
Approved Premises UI uses `parseISO` from `date-fns` to parse, and that fails on years with 2 digits (`parseISO('99-01-01') -> Invalid Date`) so I'm not sure why that check was there anyway in Approved Premises UI, but in Temporary Accommodation UI we're now using `new Date()` instead, which considers dates with 2-digit years valid, but renders those completely wrong so best avoid them:

```
new Date('01-01-01') // 2001-01-01T00:00:00.000Z, ok
new Date('99-01-01') // 1999-04-30T23:00:00.000Z, BST, ok
new Date('03-05-01') // 2001-03-05T00:00:00.000Z, lolwut?!
```